Animations/src/platform/PlatformLinux.cpp

```diff
@@ -0,0 +1,314 @@
+  char *_path = NULL;
+  char *p;
+  int result = -1;
+
+  errno = 0;
+
+  /* Copy string so it's mutable */
+  _path = strdup(path);
+  if (_path == NULL)
+    goto out;
+
+  /* Iterate the string */
+  for (p = _path + 1; *p; p++)
+  {
+    if (*p == '/')
+    {
+      /* Temporarily truncate */
+      *p = '\0';
+
+      if (maybe_mkdir(_path, mode) != 0)
+        goto out;
+
+      *p = '/';
+    }
+  }
+
+  if (maybe_mkdir(_path, mode) != 0)
+    goto out;
+
+  result = 0;
+
+out:
+  free(_path);
+  return result;
+}
+
+namespace MathAnim
+{
+  namespace Platform
+  {
+    // --------------- Internal Functions ---------------
+    // static void wideToChar(const WCHAR *wide, char *buffer, size_t bufferLength);
+    // static bool stringsEqualIgnoreCase(const char *str1, const char *str2);
+    static std::vector<std::string> availableFonts = {
+        "JetBrains Mono"};
+    static bool availableFontsCached = false;
+    static std::string homeDirectory = std::string();
+
+    const std::vector<std::string> &getAvailableFonts()
+    {
+      if (!availableFontsCached)
+      {
+        // TODO: Placeholder cuz it's 01:39
+        availableFontsCached = true;
+      }
+
+      return availableFonts;
+    }
+
+    // Adapted from https://stackoverflow.com/questions/2467429/c-check-installed-programms
+    bool isProgramInstalled(const char *displayName)
+    {
+      // TODO: Placeholder, but also this function seems unused?
+      return false;
+    }
+
+    bool getProgramInstallDir(const char *programDisplayName, char *buffer, size_t bufferLength)
+    {
+      // TODO: I'm used to Rust or managed languages and string handling in C/++ scares me
+
+      std::string const path = std::getenv("PATH");
+      char const delim = ':';
+
+      auto start = 0U;
+      auto end = path.find(delim);
+      while (true)
+      {
+        auto const folder = path.substr(start, end - start);
+
+        int i = 1;
+
+        for (auto const &directory_entry : std::filesystem::directory_iterator(folder))
+        {
+          auto const file = directory_entry.path();
+          if (strcmp(file.filename().c_str(), programDisplayName) == 0 && access(file.c_str(), X_OK) == 0)
+          {
+            strncpy(buffer, file.parent_path().c_str(), bufferLength - 1);
+            // Append / to the end of the path
+            buffer[strlen(buffer) + 1] = '\0';
+            buffer[strlen(buffer)] = '/';
+            return true;
+          }
+        }
+        if (end == std::string::npos)
+        {
+          break;
+        }
+        start = end + 1;
+        end = path.find(delim, start);
+      }
+
+      return false;
+    }
+
+    bool executeProgram(const char *programFilepath, const char *cmdLineArgs, const char *workingDirectory, const char *executionOutputFilename)
+    {
+      exit(1029);
+      std::string finalArgs = std::string("\"") + programFilepath + std::string("\" ") + cmdLineArgs;
+      // if (executionOutputFilename)
```

Comment:
Removed the commented code and the actual implementation, added a warning instead. I would like to clean this up in another PR (and implement all of the functionality properly). I want to avoid bloating this PR further.